ambuild2/frontend/v2_2/context_manager.py

```diff
@@ -0,0 +1,177 @@
+# vim: set ts=8 sts=4 sw=4 tw=99 et:
+#
+# This file is part of AMBuild.
+#
+# AMBuild is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# AMBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with AMBuild. If not, see <http://www.gnu.org/licenses/>.
+import copy
+import os
+import sys
+from ambuild2 import util
+from ambuild2.frontend import context_manager
+from ambuild2.frontend import paths
+from ambuild2.frontend.system import System
+from ambuild2.frontend.v2_2.context import BuildContext
+from ambuild2.frontend.v2_2.context import EmptyContext
+from ambuild2.frontend.v2_2.context import RootBuildContext
+from ambuild2.frontend.v2_2.context import TopLevelBuildContext
+from ambuild2.frontend.version import Version
+
+class ConfigureException(Exception):
+    def __init__(self, *args, **kwargs):
+        super(ConfigureException, self).__init__(*args, **kwargs)
```

Comment:
As a general rule of thumb, all custom exceptions have ```Error``` as the suffix, not ```Exception```
```suggestion
class ConfigureError(Exception):
    def __init__(self, *args, **kwargs):
        super(ConfigureError, self).__init__(*args, **kwargs)
```